src/test/java/TasksUtilityTest.java

```diff
@@ -0,0 +1,102 @@
+// limitations under the License.
+
+import com.google.api.services.tasks.model.Task;
+import com.google.api.services.tasks.model.TaskList;
+import com.google.appengine.repackaged.com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableList;
+import com.google.sps.model.TasksClient;
+import com.google.sps.utility.TasksUtility;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+
+/** Test Tasks Utility functions */
+@RunWith(JUnit4.class)
+public final class TasksUtilityTest {
+
+  private TasksClient tasksClient;
+
+  private static final String TASK_ID_ONE = "task one";
+  private static final String TASK_ID_TWO = "task two";
+  private static final String TASK_LIST_ID_ONE = "task list one";
+  private static final String TASK_LIST_ID_TWO = "task list two";
+
+  private static final Task TASK_ONE = new Task().setId(TASK_ID_ONE);
+  private static final Task TASK_TWO = new Task().setId(TASK_ID_TWO);
+
+  private static final TaskList TASK_LIST_ONE = new TaskList().setId(TASK_LIST_ID_ONE);
+  private static final TaskList TASK_LIST_TWO = new TaskList().setId(TASK_LIST_ID_TWO);
+
+  private static final List<TaskList> NO_TASK_LISTS = ImmutableList.of();
+  private static final List<TaskList> SOME_TASK_LISTS =
+      ImmutableList.of(TASK_LIST_ONE, TASK_LIST_TWO);
+
+  private static final List<Task> NO_TASKS = ImmutableList.of();
+  private static final List<Task> ONE_TASK = ImmutableList.of(TASK_ONE);
+  private static final List<Task> SOME_TASKS = ImmutableList.of(TASK_ONE, TASK_TWO);
+
+  @Before
+  public void setUp() {
+    tasksClient = Mockito.mock(TasksClient.class);
+  }
+
+  @Test
+  public void getAllTasksFromNoTaskLists() throws Exception {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(NO_TASK_LISTS);
+    List<Task> actual = TasksUtility.getAllTasksFromAllTaskLists(tasksClient);
+    Assert.assertEquals(ImmutableList.of(), actual);
+  }
+
+  @Test
+  public void getNoTasksFromSomeTaskLists() throws Exception {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASK_LISTS);
+    Mockito.when(tasksClient.listTasks(TASK_LIST_ONE)).thenReturn(NO_TASKS);
+    Mockito.when(tasksClient.listTasks(TASK_LIST_TWO)).thenReturn(NO_TASKS);
+    List<Task> actual = TasksUtility.getAllTasksFromAllTaskLists(tasksClient);
+    Assert.assertEquals(NO_TASKS, actual);
+  }
+
+  @Test
+  public void getSomeTasksFromSomeTaskLists() throws Exception {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASK_LISTS);
+    Mockito.when(tasksClient.listTasks(TASK_LIST_ONE)).thenReturn(ONE_TASK);
+    Mockito.when(tasksClient.listTasks(TASK_LIST_TWO)).thenReturn(SOME_TASKS);
+    List<Task> actual = TasksUtility.getAllTasksFromAllTaskLists(tasksClient);
+    Assert.assertEquals(ImmutableList.of(TASK_ONE, TASK_ONE, TASK_TWO), actual);
```

Comment:
Oh right, because we use Set now, the order might not be the same. Looks like I'd need to import [AssertJ](https://assertj.github.io/doc/) to do `assertThat(actual).containsExactlyInAnyOrder(expected);`. Would that be preferred over sorting both actual and expected?

Reply:
I just changed it to this, let me know if that works
```
Assert.assertTrue(expected.containsAll(actual));
Assert.assertTrue(actual.containsAll(expected));
```

Reply:
> Oh right, because we use Set now, the order might not be the same
Not only because we are using a Set, but even before that, the method never "promised" any order.

Works for me, the double check was a neat idea!

Reply:
Oh true you're right, I should have considered that from the very start then. Thankfully the Tasks API does return tasks and tasklists in order every time. Thanks for pointing that out!

Reply:
`RESOLVED`